forem/forem
# Add quick select ranges to date range picker

## What type of PR is this? (check all applicable)


## Description

This PR adds the final functionality to the crayons DateRangePicker component, and moves it out of the "beta" folder in storybook. We are specifically supporting the following preset time ranges:

- This month (from 1st of the current month til today)
- Last month (1st to last day of last calendar month)
- This quarter (from 1st of the current quarter til today)
- Last quarter (1st to last day of last calendar quarter)
- This year (from 1st of the current year til today)
- Last year (from 1st to last day of last calendar year)

If any of the presets fall outside of the 'max' and 'min' dates set, then they are hidden from the UI.


## Related Tickets & Documents



- Closes https://github.com/forem/forem/issues/17493

## QA Instructions, Screenshots, Recordings

- Locally run `yarn install && yarn storybook`
- Navigate to the DateRangePicker story with presets (you can also toggle the presets on/off in the default story's controls)
- Experiment with selecting preset ranges 
- Experiment with setting the min and max dates and make sure any preset outside of your max/min isn't shown

When selecting a preset, the popup calendar should close. This is to mimic the same behaviour as if the user manually clicks a start and end date.


https://user-images.githubusercontent.com/20773163/172429897-636fec7c-c8d2-451b-8d18-7616bb8a6593.mp4



### UI accessibility concerns?

I don't think so. The buttons are labelled and contained in a list. The remainder of behaviour is the same as if the user manually clicked a start and end date.

## Added/updated tests?

      have not been included_

## [Forem core team only] How will this change be communicated?

_Will this PR introduce a change that impacts Forem members or creators, the
development process, or any of our internal teams? If so, please note how you
will share this change with the people who need to know about it._

      [Storybook](https://storybook.forem.com/) (for Crayons components)
      updated. I have filled out the
      [Changes Requested](https://github.com/forem/admin-docs/issues/new?assignees=&labels=&template=changes_requested.md)
      issue template so Community Success can help update the Admin Docs
      appropriately.
      [`CHANGELOG.md`](https://github.com/forem/forem/tree/main/CHANGELOG.md)
      or in a [forem.dev](http://forem.dev) post
- [X] This change does not need to be communicated, and this is why not: not yet deployed in the app itself


## [optional] What gif best describes this PR or how it makes you feel?

![very amazing, very efficient, and very real](https://media.giphy.com/media/l4FGDs4dwl2olj1pS/giphy.gif)
